internal/sqlutil/trace.go

```diff
@@ -96,10 +96,18 @@ func trackGoID(query string) {
 	logrus.Warnf("unsafe goid %d: SQL executed not on an ExclusiveWriter: %s", thisGoID, q)
 }
 
+var dbConns = make(map[string]*sql.DB)
+
 // Open opens a database specified by its database driver name and a driver-specific data source name,
 // usually consisting of at least a database name and connection information. Includes tracing driver
 // if DENDRITE_TRACE_SQL=1
 func Open(dbProperties *config.DatabaseOptions) (*sql.DB, error) {
+	// reuse connection only if not a sqlite3 test database
+	isTestDB := strings.HasSuffix(string(dbProperties.ConnectionString), "_test.db")
+	if conn, ok := dbConns[string(dbProperties.ConnectionString)]; !isTestDB && ok {
```

Comment:
We've managed this long without leaking test-only code into the main codebase. Why is this necessary?

Reply:
Well, at least `roomserver/roomserver_test.go` reuses the same sqlite file for testing. So the approach with just using the `dbProperties.ConnectionString` as the key didn't work. (fixed now)